lib/galaxy_test/api/test_jobs.py

```diff
@@ -243,9 +244,9 @@ def test_report_error_anon(self):
         job_id = run_response['jobs'][0]["id"]
```

Comment:
Since I've introduced a check in the `JobController.error()` API to verify that the job is in error, I need to add `self.dataset_populator.wait_for_job(job_id)` here, but that fails with:
```
urllib3.connectionpool DEBUG 2020-05-12 12:26:41,891 Starting new HTTP connection (1): 127.0.0.1:8506
galaxy.web.framework.decorators ERROR 2020-05-12 12:26:41,925 Uncaught exception in exposed API method:
Traceback (most recent call last):
  File "/usr/users/ga002/soranzon/software/nsoranzo_galaxy/lib/galaxy/web/framework/decorators.py", line 282, in decorator
    rval = func(self, trans, *args, **kwargs)
  File "/usr/users/ga002/soranzon/software/nsoranzo_galaxy/lib/galaxy/webapps/galaxy/api/jobs.py", line 144, in show
    job = self.__get_job(trans, id)
  File "/usr/users/ga002/soranzon/software/nsoranzo_galaxy/lib/galaxy/webapps/galaxy/api/jobs.py", line 366, in __get_job
    return self.job_manager.get_accessible_job(trans, decoded_job_id)
  File "/usr/users/ga002/soranzon/software/nsoranzo_galaxy/lib/galaxy/managers/jobs.py", line 58, in get_accessible_job
    belongs_to_user = (job.user == trans.user) if job.user else (job.session_id == trans.get_galaxy_session().id)
AttributeError: 'NoneType' object has no attribute 'id'
urllib3.connectionpool DEBUG 2020-05-12 12:26:41,926 http://127.0.0.1:8506 "GET /api/jobs/0beccb6ec965c2fd?full=False&key=46459b20d08abf80d32d647860e67069 HTTP/1.1" 500 None
```
because it uses the test user API key not the anonymous user cookie.

Should I:
- drop this API test? (we don't have many other API tests for anonymous users any way)
- or drop the check that the job is in error?

Extending the test populators to support anonymous session seems to be a much larger project. Suggestions?

Reply:
I'd comment out the check that the job is in error until we can provide some support in populators for doing more with anonymous testing. We don't have much anonymous user testing but that is sort of a reason to not drop the tests we have in my mind.

If instead of you decide to just drop the test though I won't be -1 or anythinig... just in my mind the test is more valuable than the job in error check.

Reply:
We shouldn't remove this test, I added it because anonymous error reporting was broken. I think we may want to allow submitting reports for all job states (say someone wants to report an empty dataset, or a job that hangs in the queue or ...).

Reply:
Thanks both for the feedback, I've removed the check that the job is in error, but kept the `self.dataset_populator.wait_for_job(job_id)` call in `test_report_error()` above.